luarules/gadgets/unit_collision_damage_behavior.lua

```diff
@@ -18,52 +18,67 @@ local fallDamageMagnificationFactor = 14 --the multiplier by which default engin
 local groundCollisionDefID = Game.envDamageTypes.GroundCollision
 local objectCollisionDefID = Game.envDamageTypes.ObjectCollision
 local objectCollisionVelocityThreshold = 3.3 --this defines how fast a unit has to be moving in order to take object collision damage
-local maxImpulseThreshold = 0.08 --incease this to make units move less from impulse. This defines the max impulse damage a unit can take expressed in a percentage multiplier
-local cooldownsFrameThreshold = Game.gameSpeed --this is how long in frames before a unit's maximum impulse limit resets.
+local maxImpulseProportion = 0.04 --incease this to make units move less from impulse. This defines the max impulse damage allowed a unit can take relative to its mass.
 
-local spGetUnitVelocity = Spring.GetUnitVelocity
 local spGetUnitHealth = Spring.GetUnitHealth
-local mathmin = math.min
+local spGetUnitVelocity = Spring.GetUnitVelocity
+local spSetUnitVelocity = Spring.SetUnitVelocity
+local mathMin = math.min
+local mathMax = math.max
+local mathCeil = math.ceil
 
 local fallDamageMultipliers = {}
 local unitsMaxImpulse = {}
 local weaponDefIDImpulses = {}
-local overImpulseCooldowns = {}
-local transportingUnits = {}
+local transportedUnits = {}
 local unitMasses = {}
-
-local currentModulusFrame = 0
+local weaponDefIgnored = {}
+local unitInertiaCheckFlags = {} -- track inertia via velocity (units have constant mass)
+local gameFrame = 0
+local velocityWatchDuration = 3
 
 for unitDefID, unitDef in ipairs(UnitDefs) do
 	local fallDamageMultiplier = unitDef.customParams.fall_damage_multiplier or 1.0
-	fallDamageMultipliers[unitDefID] = fallDamageMultiplier*fallDamageMagnificationFactor
-	unitsMaxImpulse[unitDefID] = unitDef.mass/maxImpulseThreshold
+	fallDamageMultipliers[unitDefID] = fallDamageMultiplier * fallDamageMagnificationFactor
+	unitsMaxImpulse[unitDefID] = unitDef.mass / maxImpulseProportion
 	unitMasses[unitDefID] = unitDef.mass
 end
 
+for name, weaponDefID in pairs(Game.envDamageTypes or {}) do
+	weaponDefIgnored[weaponDefID] = true
+end
+
 for weaponDefID, wDef in ipairs(WeaponDefs) do
 	if wDef.damages and wDef.damages.impulseBoost and wDef.damages.impulseFactor then
 		weaponDefIDImpulses[weaponDefID] = {impulseboost = wDef.damages.impulseBoost, impulsefactor = wDef.damages.impulseFactor}
 	end
-end
 
-local function calculateImpulseData(unitDefID, damage, weaponDefID)
-	local impulse = (damage + weaponDefIDImpulses[weaponDefID].impulseboost) * weaponDefIDImpulses[weaponDefID].impulsefactor
-	local maxImpulse = unitsMaxImpulse[unitDefID]
-	local impulseMultiplier = mathmin(maxImpulse/impulse, 1)
-	return impulse, impulseMultiplier
-end
-
-local function isUnitBeingTransported(unitID)
-	for transportID, transportedID in pairs(transportingUnits) do
-		if transportingUnits[transportID][unitID] then
-			return true
+	
+	local function maxDamage(damages)
+		local damage = damages[0]
```

Comment:
Isn't this nil, since Lua tables are 1-indexed?

Reply:
no, 0 is the default armorclass

Reply:
looks like Sprunk answered this quesiton for me
I copied this code from BONELESSDRAGONHAM. It worked, so I left it.